pkg/kubectl/cmd/drain.go

```diff
@@ -282,16 +283,28 @@ func (o *DrainOptions) getController(sr *api.SerializedReference) (interface{},
 	return nil, fmt.Errorf("Unknown controller kind %q", sr.Reference.Kind)
 }
 
+func newSerializedControllerRef(pod api.Pod) *api.SerializedReference {
+	return &api.SerializedReference{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "SerializedReference",
+		},
+		Reference: api.ObjectReference{ //currently only set based on first controller
+			APIVersion: pod.GetOwnerReferences()[0].APIVersion,
```

Comment:
I suggest using a different approach (see below) that would mean this function can be removed. However, for future reference, note that `pod.GetOwnerReferences()[0]` is dangerous here because a pod can have multiple OwnerReferences, and there is no guarantee that the first one is the ControllerRef (which is the one you actually are looking for).